### PR TITLE
Add OpenAPI specifications for source cmd

### PR DESCRIPTION
**Pull request by rubhanazeem**
Co-authored-by: Eduardo Navarro <enavarro@suse.com>

- [x]  `/source?cmd=createmaintenanceincident`
- [x] `/source?cmd=branch`
- [x] `/source?cmd=orderkiwirepos`

### CODE REVIEW

File: src/api/public/apidocs-new/paths/sources_createmaintenanceincident.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+post:
+  summary: create mainatenance incident projects based on attribute search
```
Comment (dmarcoux) [inline src/api/public/apidocs-new/paths/sources_createmaintenanceincident.yaml:2]:
```suggestion
  summary: Create maintenance incident projects based on attribute search
```

File: src/api/public/apidocs-new/paths/sources_createmaintenanceincident.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  description: create mainatenance incident projects based on attribute search.
```
Comment (dmarcoux) [inline src/api/public/apidocs-new/paths/sources_createmaintenanceincident.yaml:3]: Same comment as the line above.

File: src/api/public/apidocs-new/paths/source_branch.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
+post:
+  summary: Branch a set of packages based on attributes or on existing request
+  description: Branch a set of packages based on attributes or on existing request.
```
Comment (dmarcoux) [inline src/api/public/apidocs-new/paths/source_branch.yaml:3]: What is the point of having a description if it's the same as the summary?

Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_branch.yaml]: I've updated the summaries and descriptions. :slightly_smiling_face:

File: src/api/public/apidocs-new/paths/source_branch.yaml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    Create a source link from a package of an existing project to a
+    new subproject of the requesters home project.
+  security:
+    - basic_authentication: []
+  parameters:
+    - name: project
+      description: The project that you want to branch
+      in: query
+      schema:
+        type: string
```
Comment (danidoni) [inline src/api/public/apidocs-new/paths/source_branch.yaml:13]: We have the [project name](https://github.com/openSUSE/open-build-service/blob/master/src/api/public/apidocs-new/components/parameters/project_name.yaml) for this, right?

**Review by saraycp — CHANGES_REQUESTED** [review]
I miss a few error responses too:

A _403 Forbidden_
```
<status code="modify_project_no_permission">
  <summary>no permission to modify project 'MaintenanceProject'</summary>
</status>
```
and a _400 Bad Request_

```
<status code="incident_has_no_maintenance_project">
  <summary>incident projects shall only create below maintenance projects</summary>
</status>
```

File: src/api/public/apidocs-new/paths/source_branch.yaml (hunk 4 above)
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_branch.yaml:13]: [project name](https://github.com/openSUSE/open-build-service/blob/master/src/api/public/apidocs-new/components/parameters/project_name.yaml) is different. The name of the required parameter in the API is `project` and it's a query parameter. The one you mentioned is defined as a path parameter.

File: src/api/public/apidocs-new/paths/source_createmaintenanceincident.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  description: Create a mainatenance incident project based on attribute search.
+  security:
+    - basic_authentication: []
+  parameters:
+    - in: query
+      name: attribute
+      schema:
+        type: string
+      example: OBS:Maintained
+      description: attribute used for package search, default is OBS:Maintained
```
[the remark this answers is not in the document]
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_createmaintenanceincident.yaml]: That's correct.

File: src/api/public/apidocs-new/OBS-v2.10.50.yaml [hunk 6]
```diff
@@ -17,6 +17,7 @@ tags:
   - name: Person
   - name: Published Binaries
   - name: Requests
+  - name: Sources (for global commands)
```
Comment (saraycp) [inline src/api/public/apidocs-new/OBS-v2.10.50.yaml:20]:
```suggestion
  - name: Sources
```

File: src/api/public/apidocs-new/paths/source_createmaintenanceincident.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      description: Forbidden
+      content:
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/code_and_summary.yaml'
+          example:
+            code: modify_project_no_permission
+            summary: no permission to modify project 'MaintenanceProject'
+  tags:
+    - Sources (for global commands)
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_createmaintenanceincident.yaml:46]:
```suggestion
    - Sources
```

File: src/api/public/apidocs-new/paths/source_branch.yaml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+                  attribute: true
+              summary:
+                type: string
+            xml:
+              name: status
+          example:
+            code: not_found
+            summary: no packages found by search criteria
+  tags:
+    - Sources (for global commands)
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_branch.yaml:110]:
```suggestion
    - Sources
```

File: src/api/public/apidocs-new/paths/source_orderkiwirepos.yaml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+                  attribute: true
+              summary:
+                type: string
+            xml:
+              name: status
+          example:
+            code: 400
+            summary: '/srv/obs/sources/:upload/1344: not xml'
+  tags:
+    - Sources (for global commands)
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_orderkiwirepos.yaml:54]:
```suggestion
    - Sources
```

**Review by saraycp — CHANGES_REQUESTED** [review]

File: src/api/public/apidocs-new/paths/source_orderkiwirepos.yaml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+            <repository type="rpm-md" priority="99" alias="kiwi">
+              <source path="obs://Virtualization:Appliances:Builder/openSUSE_Leap_15.3"/>
+            </repository>
+            <repository type="rpm-md" priority="1" alias="OS" imageinclude="true">
+              <source path="{exc_repo}"/>
+            </repository>
+          </image>
+  responses:
+    '200':
+      description: Sorted list repositories
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_orderkiwirepos.yaml:21]:
```suggestion
      description: Sorted list of repositories
```

**Review by saraycp — CHANGES_REQUESTED** [review]

File: src/api/public/apidocs-new/paths/source_orderkiwirepos.yaml (hunk 10 above)
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_orderkiwirepos.yaml]: Done

**Review by saraycp — APPROVED** [review]

**Pull request merged**